Feat: 이슈 등록 api & 시더 수정

yejineee opened the pull request:
# 구현사항
* 새로운 이슈를 등록하는 API 구현 (#127)

* Feat: 이슈 등록 API 구현

새로운 이슈를 등록하는 API를 구현했습니다.

* Fix: GitHub 프로필 링크 수정

기존에 설정한 프로필 링크가 프로필 사진 링크가 아닌 프로필 화면의 링크
여서 프로필 사진 링크로 등록되도록 수정했습니다.

* Fix: milestone endDate 수정

milestone의 endDate는 createdAt보다 이후여야 하므로,
createdAt 이후로 수정한다.

changheedev committed:
Refactor: Issue seeder 코드 수정 (#131)

이슈 데이터를 생성할 때 milestoneId와 isClosed를 idx 기준으로
생성하면서 같은 milestoneId에 속하는 모든 issue의 isClosed가 같은
현상이 있습니다. 그래서 open/close 상태가 적절히 섞이도록 수정해주
었습니다.

yejineee committed:
Feat: 새로운 이슈 등록 API (#134)

* 새로운 이슈를 등록하는 API 구현 (#127)

* Feat: 이슈 등록 API 구현

새로운 이슈를 등록하는 API를 구현했습니다.

* Fix: GitHub 프로필 링크 수정

기존에 설정한 프로필 링크가 프로필 사진 링크가 아닌 프로필 화면의 링크
여서 프로필 사진 링크로 등록되도록 수정했습니다.

* Fix: milestone endDate 수정

milestone의 endDate는 createdAt보다 이후여야 하므로,
createdAt 이후로 수정한다.

Co-authored-by: Changhee Choi <changhee.dev@gmail.com>
Co-authored-by: yejineee <21700431@handong.edu>

changheedev commented on server/src/libs/seeders/milestone.js:
```diff
@@ -5,7 +5,7 @@ const { Milestone: MilestoneModel } = db.sequelize.models;
 export async function up() {
   const baseTitle = 'week ';
   const baseDescription = '마일스톤 설명 ';
-  const endDates = ['2020-10-31', null, '2020-12-24'];
+  const endDates = ['2020-11-22', null, '2021-05-24'];
   const bulkMilestone = endDates.reduce((milestones, endDate, idx) => {
```
오늘 날짜 + 랜덤한 일수를 더해주면 좀 더 좋은 방식이 될 것 같습니다.

yejineee committed:
Refactor/seeder (#139)

* 새로운 이슈를 등록하는 API 구현 (#127)

* Feat: 이슈 등록 API 구현

새로운 이슈를 등록하는 API를 구현했습니다.

* Fix: GitHub 프로필 링크 수정

기존에 설정한 프로필 링크가 프로필 사진 링크가 아닌 프로필 화면의 링크
여서 프로필 사진 링크로 등록되도록 수정했습니다.

* 이슈 생성 컴포넌트 및 모델 생성  (#130)

* Feat : Sidebar 레이아웃 컴포넌트 생성

* Fix : ArrowHeader 를 ArrowContainerStyle로 변경

헤더보다는 컨테이너 자체로 사용되어 변경 함.

* Feat : 이슈 생성 페이지 content 컴포넌트 생성

* Feat : issue new Component 바디 완성 및 models 생성

* Feat : useSide 커스텀 훅 생성

사이드 영역에서 무엇이 선택이 되었는지 관리하는 훅 생성.

* Fix : 콘솔로그 제거

* Style : custom hook 네이밍 변경

useSide -> useSidebar
이름이 불분명하여 변경 함.

* Fix: milestone endDate 수정

milestone의 endDate는 createdAt보다 이후여야 하므로,
createdAt 이후로 수정한다.

* Feat: Sidebar 컴포넌트 생성 (#129)

* Feat: 마일스톤 진행률 컴포넌트 추가 (#125)

* Feat: 마일스톤의 진행률을 보여주는 유틸함수 추가

마일스톤에서 시작 시간부터 끝 시간 사이에서 현재 시간은 몇 %의 진행률인지를
보여주는 유틸리티 함수를 추가하였다.
현재시간이 끝시간보다 크다면, 100을 반환한다.
시작시간이 끝 시간보다 크다면, 에러를 반환한다.
위 두 경우를 제외하고는 진행률을 % 단위로 반환한다.

* Test: 마일스톤 진행률 테스트 추가

마일스톤 진행률을 보여주는 유틸리티 함수의 테스트를 추가하였다.

* Feat: 마일스톤 진행률 컴포넌트 추가

시작시간과 끝 시간이 주어졌을 때, 현재 시간상 진행률을 보여주는
컴포넌트를 추가하였다.

Co-authored-by: yejineee <21700431@handong.edu>

* Feat: sidebar 레이아웃 구현 (#126)

사이드바 구현에 필요한 컴포넌트 구현.

* 마크다운 에디터 상태 분리 (#124)

* Refactor : 마크다운 에디터 상태 분리

* Fix: milestone dummy data field 이름 수정

실제 api 구현 시 endDate로 데이터가 오기 때문에,
더미데이터도 이와 맞춰 end_date로 되어있는 것을 endDate로 수정한다.

* Fix: milestone progress를 실제 데이터를 props로 받는 것으로 수정

milestone을 가져오는 api를 연결하여 실제 데이터를 가져왔으므로,
더미데이타를 사용했던 것을 props로 가져온 데이터를 사용하는 것으로 수정한다.

* Chore: Rename TextDropdownItem -> FilterDropdownItem

TextDropDownItem을 필터와 마일스톤 두 곳에서 범용적으로
사용하는데 문제가 있어, 필터 드롭다운으로 이름을 변경한다.

* Style: dropdown을 button으로 수정

dropdown 아이템을 클릭했을 시, 클릭 이벤트가 발생해야 하므로
DropDownItemContainer를 button으로 수정한다.
버튼으로 변경된 후, 스타일을 조정하였다.

* Fix: 드롭다운에 click handler, dataset-id, 조건부 아이콘 렌더링

각 드롭다운마다 고유한 id가 무엇인지 설정하여, 클릭 이벤트 발생 시
이벤트 핸들러에서 id를 알 수 있도록 설정하였다.
또한, 드롭다운마다 클릭 이벤트 핸들러를 설정하였다.
드롭다운이 선택되었을 경우에만, 체크 아이콘이 보이도록 설정하였다.

* Feat: 프론트에서 서버로 데이터 가져오는 api 생성

마일스톤, 라벨, 유저 리스트를 프론트에서 서버로 데이터를 가져오는
api를 추가하였다. 이는 사이드바에서 사용이 된다.

* Feat: 마일스톤, 유저, 라벨이 선택된 여부에 따라 렌더링

마일스톤과 유저, 라벨은 선택되지 않았을 경우
no milestones와 같은 메시지를 띄운다.
만약 선택된 것이 있는 경우, 선택된 것을 렌더링해주도록 한다.

* Feat: 사이드바 컴포넌트 구성

assignee, labels, milestones 세 가지의 요소를 합쳐
사이드바를 생성하는 사이드바 컴포넌트를 구현하였다.

* Feat: 사이드바 모델 생성

사이드바 모델에서는 labels, assignees, milestones를
서버로부터 가져와서 사이드바 뷰에 데이터를 props로 넘겨준다.

* Feat: 사이드바에서 fetch 성공시 리듀서, action-type 추가

사이드바에서는 label, assignee, milestone을 갖고 있다.
각 경우마다 fetch에 성공했을 시, state에 추가하는 리듀서를 구현하였다.

* Fix: console.log 삭제

디버깅용으로 있던 console.log를 삭제한다.

* Style: sidebar border 및 dropdown item border 수정

사이드바의 외부 border는 디버깅 용이므로 삭제하고,
사이드바 컴포넌트 사이에 border-top을 주었다.
dropdown item에서도 자식 컴포넌트 사이에 border-top을 주었다.
위의 결과로 dropdown header와 dropdown 아이템 사이에 구분선이 필요하여,
dropdown header의 border-bottom을 추가하였다.

* Fix: sidebar index.js에서 api 호출 시  불필요한 클로저 함수 삭제

인자값을 받는 경우가 아니면, 클로저를 만들 이유가 없기 때문에,
바로 api 함수를 넘겨주는 것으로 수정하였다.

* Refactor: sidebar component 분리

사이드바에서 선택된 담당자, 라벨, 마일스톤을 렌더링하는 부분이
복잡하므로, 상위의 컨테이너로 분리시켰다.

* Style: dropdown profile 사이즈 변경

dropdown의 유저 프로필이 타원형이었다. 이를 수정하기 위해
유저 프로필의 크기를 줄이고, dropdown의 패딩 상하부분을 줄였다.

* Fix: dropdown 안에 있는 click handler에 id 전달

기존에는 dataset에 dropdown item의 id를 저장하였으나,
이는 html에 직접 드러나는 문제가 있어서, 클릭 핸들러의 인자로 id를 전달하는 것으로 수정한다.

Co-authored-by: yejineee <21700431@handong.edu>
Co-authored-by: LeeSuKyeong <46044132+LeeSuKyeong@users.noreply.github.com>
Co-authored-by: ParkSeungHwan <44409642+rolled-potatoes@users.noreply.github.com>

* 이슈 상세 페이지 Viewer 변경 및 Comment 컴포넌트 스타일 수정 (#137)

* Fix: 중복된 내용의 파일 제거

이슈 디테일 페이지 컴포넌트에서 중복된 내용의 파일이 존재하여 삭제함.

* Fix: Comment 컴포넌트 레이아웃 틀어짐 수정 및 레이아웃 개선

새로 merge된 커밋으로 인해 Comment 컴포넌트의 레이아웃이 틀어지는
문제가 있었고 해당 문제를 수정했습니다.
수정과정에서 반응형 처리등 레이아웃을 개선시키는 작업을 했습니다.

* Refactor: 이슈 상세페이지에 SidebarLayout 컴포넌트 적용

이슈 상세페이지 레이아웃을 SidebarLayout 컴포넌트로 변경해주었습니다.

* Refactor: 랜덤 날짜 생성 함수 utils에 추가

기존에 랜덤 날짜 생성하는 함수를 여러 곳에서 쓰이기 때문에 utils에
넣도록 한다.

* Fix: 마일스톤 종료시간을 랜덤하게 설정

마일스톤의 종료시간을 현재시간부터 랜덤한 시간으로 정하였다.

Co-authored-by: Changhee Choi <changhee.dev@gmail.com>
Co-authored-by: ParkSeungHwan <44409642+rolled-potatoes@users.noreply.github.com>
Co-authored-by: yejineee <21700431@handong.edu>
Co-authored-by: LeeSuKyeong <46044132+LeeSuKyeong@users.noreply.github.com>

changheedev committed:
Revert "Refactor/seeder (#139)" (#141)

This reverts commit a84cbaf35377296ef3c4ec5552d1e2a4fae7a787.
